### PR TITLE
Support for Jaya's cucushift #8988

### PR DESCRIPTION
**Pull request by xingxingxia**
This is framework support for https://github.com/openshift/cucushift/pull/8988's upcoming fix change (i.e. [this review suggestion](https://github.com/openshift/cucushift/pull/8988#discussion_r779550497)), which needs several times of curl watches in background and needs check them.

### CODE REVIEW

**Comment by xingxingxia** [comment]
@jmekkatt @wangke19 @gangwgr @liangxia @pruan-rht help review / merge, thx!

**Comment by jmekkatt** [comment]
I have tested the change w.r.t https://github.com/openshift/cucushift/pull/8988#issuecomment-1009779566 and it is successful.
/lgtm

File: features/step_definitions/cli.rb [hunk 1]
```diff
@@ -198,9 +198,15 @@
   @result = @bg_processes.last.result
 end
 
-Given /^I check status of last background process$/ do
+Given /^I check status of( and pop)? last background process$/ do |pop|
```
Comment (liangxia) [inline features/step_definitions/cli.rb:201]: Perhaps `I check status( and pop)? of last background process` ?

File: features/step_definitions/cli.rb [hunk 2]
```diff
@@ -198,9 +198,15 @@
...
   @bg_processes.last.wait(10)
   @result = @bg_processes.last.result
+  if pop
+    raise "0 background process left" unless @bg_processes.length > 0
```
Comment (wangke19) [inline features/step_definitions/cli.rb:205]: `@bg_processes.length` should >= 0, seems the condition always is unable to meet. Replace with `if  @bg_processes.length == 0`, the raise will be executed.

Comment (xingxingxia) [reply features/step_definitions/cli.rb:205]: @wangke19 `unless @bg_processes.length > 0` is equivalent to `if @bg_processes.length == 0`, so I prefer to not change it.

File: features/step_definitions/cli.rb (hunk 1 above)
Comment (liangxia) [reply features/step_definitions/cli.rb:201]: @xingxingxia Forget it. Let's leave it as it is now.

Comment (pruan-rht) [reply features/step_definitions/cli.rb:201]: @xingxingxia the phrasing of this step is confusing.  Do you mean what does `pop` mean here?

Comment (xingxingxia) [reply features/step_definitions/cli.rb:201]: The `pop` is the `push` opposite, which means removing an entry from one stack-like object like the `@bg_processes` array.

**Pull request merged**